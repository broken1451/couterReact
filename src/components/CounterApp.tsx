import React, { Fragment, useState } from "react";
import PropTypes from "prop-types";

const CounterApp = ({ value = 0 }: any) => {
  // hooks
  const [counter, setCounter] = useState(value); // devuelve un arreglo []
  console.log({ counter, setCounter });

  // add
  const add = () => {
    // setCounter(counter + 1)
    setCounter(counter + 1);
    // setCounter((c) => {
    //   // console.log({c});
    //   // c es el valor anterior del conuter
    //   return c + 1;
    // });
    // console.log(e)
  };

  const less = () => {
    setCounter(counter - 1);
  };

  const reset = () => {
    setCounter(value);
  };

  return (
    <>
      <p>Counter App </p>
      <p> {counter} </p>
      <button onClick={add}>+1</button>
      <button onClick={reset}>Reset</button>
      <button onClick={less}>-1</button>
    </>
  );
};

CounterApp.propTypes = {
  value: PropTypes.number, // si no se coloca nada es opcional
};

export default CounterApp;
// rafcp - crear el componente
