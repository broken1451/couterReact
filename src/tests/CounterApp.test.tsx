import React from "react";
import "@testing-library/jest-dom";
import { shallow } from "enzyme";
import "@testing-library/jest-dom";
import CounterApp from "../components/CounterApp";

describe("Pruebas en el archivo counter", () => {
  // test('debe de mostrar el mensaje "Hola soy goku" ', () => {
  //     // const saludo = 'Hola soy goku';
  //     // const wrapper = render(<PrimeraApp saludo={saludo}/>) // componente de react, producto renderizado del componente de react
  //     // wrapper.getByText(saludo)
  //     // // const {getByText} = render(<PrimeraApp saludo={saludo} />) // componente de react, producto renderizado del componente de react
  //     // // const {getByText} = render(<PrimeraApp saludo={saludo} />) // componente de react, producto renderizado del componente de react
  //     // // expect( getByText(saludo)).toBeInTheDocument();
  //     // expect( wrapper.getByText(saludo)).toBeInTheDocument();
  // });

  let wrapper = shallow(<CounterApp />);

  beforeEach(() => {
    // console.log('before each');
    wrapper = shallow(<CounterApp />);
  });

  test("debe de mostrar <CounterApp /> correctamente con sus valores por defecto con el snapshot", () => {
    expect(wrapper).toMatchSnapshot();
  });

  test("debe de mostrar el valor por defecto de 100 ", () => {
    const value = 100;
    const wrapper = shallow(<CounterApp value={value} />);
    const textParrafo = wrapper.find("#counter").text().trim();
    expect(textParrafo).toBe(String(value));
  });

  test("debe de incrementar con el boton +1", () => {
    const btn1 = wrapper.find("button").at(0).simulate("click"); // rescartar las pociciones de etiquetas html dependiendo del indice
    // console.log({btn1: btn1.html()});
    const countertext = wrapper.find("#counter").text().trim();
    expect(countertext).toBe("11");
  });

  test("debe de devcrementar con el boton -1", () => {
    // const wrapper = shallow(<CounterApp />)
    const btn3 = wrapper.find("button").at(2).simulate("click"); // rescartar las pociciones de etiquetas html dependiendo del indice
    // console.log({btn3: btn3.html()});
    const countertext = wrapper.find("#counter").text().trim();
    expect(countertext).toBe("9");
  });

  test("debe de colocolar el valor por defecto con el boton reset", () => {
    const wrapper = shallow(<CounterApp value={105} />);
    wrapper.find("button").at(0).simulate("click");
    wrapper.find("button").at(0).simulate("click");
    const btn2 = wrapper.find("button").at(1).simulate("click"); // rescartar las pociciones de etiquetas html dependiendo del indice
    const countertext = wrapper.find("#counter").text().trim();
    // console.log({countertext});
    expect(countertext).toBe("105");
  });
});
