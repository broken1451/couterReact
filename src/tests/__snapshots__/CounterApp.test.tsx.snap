// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el archivo counter debe de mostrar <CounterApp /> correctamente con sus valores por defecto con el snapshot 1`] = `
<Fragment>
  <p>
    Counter App 
  </p>
  <p
    id="counter"
  >
    10
  </p>
  <button
    onClick={[Function]}
  >
    +1
  </button>
  <button
    onClick={[Function]}
  >
    Reset
  </button>
  <button
    onClick={[Function]}
  >
    -1
  </button>
</Fragment>
`;
